hub/core/chunk.py

```diff
@@ -110,29 +193,65 @@ def register_sample_to_headers(
             ValueError: If `incoming_num_bytes` is not divisible by `num_samples`.
         """
 
-        num_bytes_per_sample = incoming_num_bytes
         self.shapes_encoder.register_samples(sample_shape, 1)
-        self.byte_positions_encoder.register_samples(num_bytes_per_sample, 1)
+        if (
+            incoming_num_bytes is not None
```

Comment:
why would incoming_num_bytes be None?

Reply:
i feel like with all this special casing, it might be cleaner to use polymorphism for chunks.

something like:

- `Chunk` baseclass
- `SampleCompressedChunk`
- `CompressedChunk`

maybe?

Reply:
Agreed, would make the code much more readable

Reply:
Incoming bytes doesnt make sense for chunkwise image compression.